clang/test/CheckedCRewriter/stmtExpr.c

```diff
@@ -0,0 +1,23 @@
+// RUN: cconv-standalone -alltypes %s -- | FileCheck -match-full-lines -check-prefixes="CHECK_ALL","CHECK" %s
+// RUN: cconv-standalone %s -- | FileCheck -match-full-lines -check-prefixes="CHECK_NOALL","CHECK" %s
+// RUN: cconv-standalone %s -- | %clang -c -fcheckedc-extension -x c -o /dev/null - 
+
+typedef unsigned long size_t;
+extern _Itype_for_any(T) void *malloc(size_t size) : itype(_Array_ptr<T>) byte_count(size);
+
+void foo2(int *x) { 
+    int *p;
+    //CHECK_ALL: _Array_ptr<int> p : count(3) = ((void *)0);
+    //CHECK_NOALL: int *p;
+    p = ({int *q = malloc(3*sizeof(int)); q[2] = 1; q;});
+    //CHECK_ALL: p = ({_Array_ptr<int> q : count(3) =  malloc<int>(3*sizeof(int)); q[2] = 1; q;});
+    //CHECK_NOALL: p = ({int *q = malloc<int>(3*sizeof(int)); q[2] = 1; q;});
+    p[1] = 3;
+} 
+
+/*ideally, q would also be marked as checked, but when the decl and init are on the same, the rewriter fails*/
```

Comment:
This is a broken result -- the rewritten code won't compile. If you can't make `p` checked then you should not make `q` checked either.

Is it a rewriter bug that `q` is not made checked?

Reply:
I see that you have an issue #195 for this? If so, maybe make `foo` a separate file and name it `stmtExprBUG.c` to emphasize that it's the wrong output.

Reply:
You're right on both counts - it's a rewriter bug that results in q not being checked, and #195 is the issue set to track this :)
As it turns out, in the no alltypes case (which is where we compile), `p` doesn't get converted (because `p` is an `_Array_ptr`, so it gets made WILD), which means the rewritten code does compile (with our version of clang, anyway). But in either case, I ought to move this portion of the code to a separate file anyway to indicate that there's a bug that needs to be fixed. (I have a good feeling John's refactoring of the rewriting process might actually do the trick - our solving system is all good w.r.t making `q` checked).

In my latest commit, I split `stmtExpr.c` into two tests : `stmtExprA.c` and `stmtExprBUG.c. I've made a note to myself to remerge them once rewriting is fixed so we don't have two identical tests later on.